parser/src/main/java/com/distkv/dst/parser/DstNewSqlListener.java

```diff
@@ -134,13 +136,51 @@ public void enterListRGet(DstNewSQLParser.ListRGetContext ctx) {
   }
 
   @Override
-  public void enterListDelete(DstNewSQLParser.ListDeleteContext ctx) {
-    // TODO(qwang): Refine.
+  public void enterListRemove(DstNewSQLParser.ListRemoveContext ctx) {
+    Preconditions.checkState(parsedResult == null);
+    Preconditions.checkState(ctx.children.size() == 2);
+
+    final ParseTree listRemoveArgumentsParseTree = ctx.children.get(1);
+    final int numArguments = listRemoveArgumentsParseTree.getChildCount();
+    ListProtocol.RemoveRequest.Builder builder = ListProtocol.RemoveRequest.newBuilder();
+    final String key = listRemoveArgumentsParseTree.getChild(0).getText();
+    builder.setKey(key);
+
+    if (2 == numArguments) {
+      // RemoveOne
+      Preconditions.checkState(listRemoveArgumentsParseTree.getChildCount() == 2);
+
+      builder.setType(ListProtocol.RemoveType.RemoveOne);
+      builder.setIndex(Integer.valueOf(listRemoveArgumentsParseTree.getChild(1).getText()));
+    } else if (3 == numArguments) {
+      // RemoveRange
+      Preconditions.checkState(listRemoveArgumentsParseTree.getChildCount() == 3);
+
+      builder.setType(ListProtocol.RemoveType.RemoveRange);
+      builder.setFrom(Integer.valueOf(listRemoveArgumentsParseTree.getChild(1).getText()));
+      builder.setEnd(Integer.valueOf(listRemoveArgumentsParseTree.getChild(2).getText()));
+    } else {
+      throw new RuntimeException("Failed to parser the command.");
```

Comment:
switch case `default`

Reply:
I didn't violate the coding norm. I think it doesn't matter~